rust/cymbal/src/issue_resolution.rs

```diff
@@ -0,0 +1,86 @@
+use uuid::Uuid;
+
+use crate::error::UnhandledError;
+
+pub struct IssueFingerprintOverride {
+    pub id: Uuid,
+    pub team_id: i32,
+    pub fingerprint: String,
+    pub issue_id: Uuid,
+    pub version: i64,
+}
+
+#[derive(Debug, sqlx::FromRow)]
+pub struct Issue {
+    pub id: Uuid,
+    pub team_id: i32,
+    pub fingerprint: String,
+}
+
+pub async fn load_issue_override<'c, E>(
+    executor: E,
+    team_id: i32,
+    fingerprint: &str,
+) -> Result<Option<IssueFingerprintOverride>, UnhandledError>
+where
+    E: sqlx::Executor<'c, Database = sqlx::Postgres>,
+{
+    Ok(sqlx::query_as!(
+        IssueFingerprintOverride,
+        r#"
+            SELECT id, team_id, fingerprint, issue_id, version
+            FROM posthog_errortrackingissuefingerprintv2
+            WHERE team_id = $1 AND fingerprint = $2
+        "#,
+        team_id,
+        fingerprint
+    )
+    .fetch_optional(executor)
+    .await?)
+}
+
+pub async fn create_issue<'c, A>(
```

Comment:
The way this function is written, you'll create extra issues, since you'll create the `errortrackingissue` row even in cases where the `errortrackingissuefingerprintv2` insert hits a conflict. I think we need to think a bit harder about how these queries will work (I definitely do), will try to circle back with something concrete in the morning.

Reply:
It'd be cool not to use a transaction at all, but yeah we totally can, will modify the code to do that.